src/hotspot/share/code/nmethod.cpp

```diff
@@ -1560,6 +1706,7 @@ nmethod::nmethod(
       memcpy(speculations_begin(), speculations, speculations_len);
     }
 #endif
+    memset(immutable_data_references_begin(), 1, oopSize);
```

Comment:
Don't use `memset`, use nmethod's inline function to set counter which use correct cast to `int`.